GenderPayGap.WebUI/Controllers/Admin/AdminOrganisationCompanyNumberController.cs

```diff
@@ -135,23 +135,23 @@ private IActionResult Change(Organisation organisation, AdminOrganisationCompany
 
             string formattedCompanyNumber = viewModel.NewCompanyNumber.Trim().ToUpper();
 
-            if (formattedCompanyNumber == organisation.CompanyNumber?.Trim()?.ToUpper())
+            if (formattedCompanyNumber == organisation.CompanyNumber?.Trim().ToUpper())
             {
                 viewModel.AddErrorFor(m => m.NewCompanyNumber,
                     "Company number must be different to the current company number");
                 return View("ChangeOrganisationCompanyNumber", viewModel);
             }
 
+            // Check that another stored organisation doesn't have the same company number
             Organisation organisationWithSameNewCompanyNumber =
                 dataRepository.GetAll<Organisation>()
                     .FirstOrDefault(o => o.CompanyNumber == formattedCompanyNumber);
 
             if (organisationWithSameNewCompanyNumber != null)
             {
-                viewModel.AddErrorFor<AdminOrganisationCompanyNumberViewModel, string>(
+                viewModel.AddErrorFor(
                     m => m.NewCompanyNumber,
-                    $"Another organisation ({organisationWithSameNewCompanyNumber.OrganisationName}, "
-                    + $"with id {organisationWithSameNewCompanyNumber.OrganisationId}) has this company number");
+                    $"Another organisation ({organisationWithSameNewCompanyNumber.OrganisationName}) has this company number");

```

Comment:
Removed ID from the error message here as it looked a bit weird - agreed this with Claire :)